runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -252,405 +252,256 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+   // Constant value of 0x20, used to convert between upper and lower
+   generateVRIaInstruction(cg, TR::InstOpCode::VREPI, node, charOffsetVector, static_cast<uint16_t>(0x20), elementSizeMask);
 
+   generateRRInstruction(cg, TR::InstOpCode::LR, node, loadLength, lengthRegister);
 
-   /* ===================== Step 2: Length calculations (rResidue and 16-byte alignment) =====================*/
-
-
-   // Add string val offset to save additions in core loop
-   cursor =                  generateRXInstruction          (cg, TR::InstOpCode::getLoadAddressOpCode(), node, rSrcVal, generateS390MemoryReference(rSrcVal, NULL, valueContentOffset, cg), cursor);    iComment("source.val += strValOffset + source.offset");
-   cursor =                  generateRXInstruction          (cg, TR::InstOpCode::getLoadAddressOpCode(), node, rTgtVal, generateS390MemoryReference(rTgtVal, valueContentOffset, cg), cursor);          iComment("tgt.val += strValOffset");
-
-   // The core loop works in multiples of 16 so we need to align the length to 16, and also load the rResidue so we can operate on them separately
-   if(!isCompressedString)
-      {
-       // convert num_of_char to num_of_bytes we need to process
-      cursor =               generateRSInstruction          (cg, TR::InstOpCode::getShiftLeftLogicalSingleOpCode(), node, rLen, rLen, 1);                                                                  iComment("string length : char -> bytes");
-      }
-   cursor =                  generateRRInstruction          (cg, TR::InstOpCode::getLoadRegOpCode(), node, rResidue, rLen);
-   cursor =                  generateRILInstruction         (cg, TR::InstOpCode::NILF, node, rResidue, 0xF);                                                                                                 iComment("calc residue (can eq length if length < 16)");
-   cursor =                  generateS390BranchInstruction  (cg, TR::InstOpCode::BRC , TR::InstOpCode::COND_BZ, node, coreLoopSetup);                                                                            iComment("len perfect multiple of 16, goto main loop (rare)");
-
-   // We already set start ICF at the branch after the TMLH so don't do it here, to prevent nested ICF issues
-   if (is64)
-      cursor->setStartInternalControlFlow();
-
-   // VLL and VSTL are provided with an index not with a count (length) so we need to offset by 1.
-   cursor =                 generateRILInstruction          (cg, TR::InstOpCode::getSubtractLogicalImmOpCode(), node, rResidue, 1);                                                                         iComment("adjust residue for VLL/VSTL (length -> index)");
-
-   /* ===================== Step 3: Residue processing =====================*/
+   generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, loadLength, lengthRegister);
```

Comment:
This seems unnecessary given the load above. Can we remove this? Also shouldn't `lengthRegister` always a 32-bit value? Shouldn't we always use an `LR` here? Similarly to above, why can't `loadLength` also be a 32-bit register?

Reply:
cf269b9